### DISCUSSION TITLE
Need help with Git language detection

### DISCUSSION THREAD

**Discussion by Sermio** [community/community#141047] [Programming Help]
### Body

Hi,

I've realized that Git is detecting Makefile as the main language, but when i click it to check which files are being detected, they are not shown at all. The project is made in Dart, so it should be the main language.
![image](https://github.com/user-attachments/assets/8f52a029-ca16-4cb2-ab7f-c4223bf4476c)
![image](https://github.com/user-attachments/assets/cafdb989-b350-43ca-9a49-4bad74c44fd5)

Thanks!

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by ricardious** [comment]
### Inglés:
Hi, sorry if my English is not good. To solve the issue with Git detecting `Makefile` as the main language instead of Dart, you can try the following:

1. **Check your `.gitattributes` file**: Add or update the `.gitattributes` file in your repository to specify Dart as the main language. Add this line:
   
   ```gitattributes
   *.dart linguist-language=Dart
   ```

This makes GitHub think Dart is the main language.

**Reply by ricardious** [reply]
If it helped you, don't forget, I hope you mark my answer as correct.

**Reply by MOHAMED-LAAGUILI** [reply]
*.dart linguist-language=Dart

**Reply by Sermio** [reply]
> ### Inglés:
> 
> Hi, sorry if my English is not good. To solve the issue with Git detecting `Makefile` as the main language instead of Dart, you can try the following:
> 
>     1. **Check your `.gitattributes` file**: Add or update the `.gitattributes` file in your repository to specify Dart as the main language. Add this line:
>        ```gitattributes
>        *.dart linguist-language=Dart
>        ```
> 
> 
> This makes GitHub think Dart is the main language.

Hi, thanks for the response.

I tried that, i added the gitattributes file to the repo and it's still not showing the corret language. I'll link you my repo so that you can take a look in case i am doing something wrong!

repo: https://github.com/Sermio/WD_filter/tree/main
